sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/producer_client.hpp

```diff
@@ -40,6 +42,20 @@ namespace Azure { namespace Messaging { namespace EventHubs {
     /**@brief  The maximum size of the message that can be sent.
      */
     Azure::Nullable<std::uint64_t> MaxMessageSize{};
+
+  private:
+    friend class ProducerClient;
```

Comment:
Why is it necessary to make this a friend?

Reply:
This is so that `ProducerClient` can access the value of `CppStandardVersion` when invoking `SetUserAgent()`, but no other class can access it, or change it.